### PR TITLE
Refactor notifications to use Protobuf schemas

### PR DESCRIPTION
**Pull request by nscuro**
> **Note**  
> This PR is based on #64

### Description

Companion PR to https://github.com/DependencyTrack/hyades/pull/415

### Addressed Issue

N/A

### Additional Details

N/A

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
- ~This PR fixes a defect, and I have provided tests to verify that the fix is effective~
- [ ] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- ~This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)~
- ~This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly~

### CODE REVIEW

File: src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+            case BOM_CONSUMED -> GROUP_BOM_CONSUMED;
+            case BOM_PROCESSED -> GROUP_BOM_PROCESSED;
+            case VEX_CONSUMED -> GROUP_VEX_CONSUMED;
+            case VEX_PROCESSED -> GROUP_VEX_PROCESSED;
+            case POLICY_VIOLATION -> GROUP_POLICY_VIOLATION;
+            case PROJECT_CREATED -> GROUP_PROJECT_CREATED;
+        };
+    }
+
+    private static Optional<com.google.protobuf.Any> convert(final Object subject) {
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java:146]: That is intended here.

File: src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+                .setProject(convert(analysis.getProject()))
+                .setVulnerability(convert(analysis.getVulnerability()))
+                .setSuppressed(analysis.isSuppressed());
+
+        Optional.ofNullable(analysis.getAnalysisState()).map(Enum::name).ifPresent(builder::setState);
+
+        return builder.build();
+    }
+
+    private static Vulnerability.Cwe convert(final org.dependencytrack.model.Cwe cwe) {
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java:353]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+
+        return builder.build();
+    }
+
+    private static PolicyViolation convert(final org.dependencytrack.model.PolicyViolation policyViolation) {
+        return PolicyViolation.newBuilder()
+                .setUuid(policyViolation.getUuid().toString())
+                .setType(policyViolation.getType().name())
+                .setTimestamp(Timestamp.newBuilder()
+                        .setSeconds(policyViolation.getTimestamp().getTime() / 1000))
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java:306]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/search/IndexManager.java [hunk 4]
```diff
@@ -248,19 +247,23 @@ public void commit() {
...
     protected void handleCorruptIndexException(CorruptIndexException e) {
         LOGGER.error("Corrupted Lucene index detected", e);
-        Notification.dispatch(new Notification()
-                .scope(NotificationScope.SYSTEM)
-                .group(NotificationGroup.INDEXING_SERVICE)
-                .title(NotificationConstants.Title.CORE_INDEXING_SERVICES + "(" + indexType.name().toLowerCase() + ")")
-                .content("Corrupted Lucene index detected. Check log for details. " + e.getMessage())
-                .level(NotificationLevel.ERROR)
-        );
+        NotificationUtil.dispatchExceptionNotifications(
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/search/IndexManager.java:261]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+                .setProject(convert(analysis.getProject()))
+                .setVulnerability(convert(analysis.getVulnerability()))
+                .setSuppressed(analysis.isSuppressed());
+
+        Optional.ofNullable(analysis.getAnalysisState()).map(Enum::name).ifPresent(builder::setState);
+
+        return builder.build();
+    }
+
+    private static PolicyViolationAnalysis convert(final org.dependencytrack.model.ViolationAnalysis analysis) {
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java:341]: @sonatype-lift ignore

File: src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+
+    private static Policy convert(final org.dependencytrack.model.Policy policy) {
+        return Policy.newBuilder()
+                .setUuid(policy.getUuid().toString())
+                .setName(policy.getName())
+                .setViolationState(policy.getViolationState().name())
+                .build();
+    }
+
+    private static VulnerabilityAnalysis convert(final org.dependencytrack.model.Analysis analysis) {
```
[the remark this answers is not in the document]
Comment (nscuro) [reply src/main/java/org/dependencytrack/parser/hyades/NotificationModelConverter.java:329]: @sonatype-lift ignore

**Review by VithikaS — APPROVED** [review]

**Pull request merged**